core/src/main/java/com/netflix/conductor/service/WorkflowBulkServiceImpl.java

```diff
@@ -143,4 +150,43 @@ public BulkResponse terminate(List<String> workflowIds, String reason) {
         }
         return bulkResponse;
     }
+
+    /**
+     * Terminate workflows execution.
+     * @param workflowIds - list of workflow Ids  to perform terminate operation on
+     * @param archiveWorkflow Archives the workflow
+     * @return bulk response object containing a list of succeeded workflows and a list of failed ones with errors
+     */
+    @Service
+    public BulkResponse delete(List<String> workflowIds, boolean archiveWorkflow) {
+        BulkResponse bulkResponse = new BulkResponse();
+        for (String workflowId : workflowIds) {
+            try {
+                executionDAOFacade.removeWorkflow(workflowId, archiveWorkflow);;
+                bulkResponse.appendSuccessResponse(workflowId);
+            } catch (Exception e) {
+                LOGGER.error("bulk terminate exception, workflowId {}, message: {} ",workflowId, e.getMessage(), e);
+                bulkResponse.appendFailedResponse(workflowId, e.getMessage());
+            }
+        }
+        return bulkResponse;
+    }
+
+    @Service
+    public BulkResponse removeCorrelatedWorkflows(String correlationId, boolean archiveWorkflow) {
+        BulkResponse bulkResponse = new BulkResponse();
+        String query = String.format("correlationId=\"%s\"", correlationId);
+        SearchResult<String> result = executionDAOFacade.searchWorkflows(query, "*", 0, 5000, null);
+        result.getResults().stream().parallel().forEach(workflowId -> {
+            try {
+                executionDAOFacade.removeWorkflow(workflowId, archiveWorkflow);;
+                bulkResponse.appendSuccessResponse(workflowId);
+            } catch (Exception e) {
+                LOGGER.error("bulk terminate exception, workflowId {}, message: {} ", workflowId, e.getMessage(), e);
```

Comment:
Does logging exception, log the entire traceback? Due to polling we could be repeatedly trying to remove a workflow and if the removal fails, we could end up with a lot of logs. Please be carefuly with conductor logging

Reply:
No, not the entire trace back.